### PR TITLE
code editing through streams

### PR DESCRIPTION
**Pull request by supercrafter100**
This PR is heavily inspired by #61 and uses some code of it. It however still uses streaming to have that low latency start. All credits for figuring out the Diff parts go to @dipamsen I would've never figured that one out. The only thing I really did was incorperate it in the streaming system so we could keep using streams for the prompts instead of having the latency of waiting for the entire response.

### CODE REVIEW

File: src/util/realisticTyping.js [hunk 1]
```diff
@@ -34,20 +34,55 @@ const typeRealistically = async (editor, code, delay = 50) => {
...
+      await typeRealistically(editor, diff.value);
+    } else if (diff.removed) {
+      // delete diff.count characters
+      const position = editor.selection.active;
+      const newPosition = move(position, diff.value);
+      const range = new vscode.Range(position, newPosition);
+      await sleep(10 * diff.count);
+      editor.selection = new vscode.Selection(position, newPosition);
+      scrollToCursor(editor);
+      await sleep(200);
```
Comment (shiffman) [inline src/util/realisticTyping.js:65]: should we make this a variable so we can adjust the speed?

File: src/util/realisticTyping.js [hunk 2]
```diff
@@ -34,20 +34,55 @@ const typeRealistically = async (editor, code, delay = 50) => {
...
+      });
+      scrollToCursor(editor);
+    } else {
+      // shift cursor to end of diff
+      const position = editor.selection.active;
+      const newPosition = move(position, diff.value);
+      await sleep(100);
+      editor.selection = new vscode.Selection(newPosition, newPosition);
+      scrollToCursor(editor);
+      await sleep(100);
```
Comment (shiffman) [inline src/util/realisticTyping.js:77]: same here, variable to adjust speed? all the speed can be one single value and everything relative to it.

**Review by shiffman — COMMENTED** [review]
Amazing!

File: src/util/realisticTyping.js (hunk 1 above)
Comment (dipamsen) [reply src/util/realisticTyping.js:65]: yeah this is a great tunable parameter, we can play with it to make it most "seamless"

**Comment by shiffman** [comment]
we can add the delay variable in another branch!

**Pull request merged**